src/main/java/org/sonatype/nexus/ci/nxrm/v3/CreateTagStep.java

```diff
@@ -0,0 +1,220 @@
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.nxrm.v3;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+import com.sonatype.nexus.api.exception.RepositoryManagerException;
+import com.sonatype.nexus.api.repository.v3.RepositoryManagerV3Client;
+import com.sonatype.nexus.api.repository.v3.Tag;
+
+import org.sonatype.nexus.ci.util.FormUtil;
+import org.sonatype.nexus.ci.util.NxrmUtil;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import jenkins.tasks.SimpleBuildStep;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import static com.sonatype.nexus.api.common.ArgumentUtils.checkArgument;
+import static com.sonatype.nexus.api.common.NexusStringUtils.isBlank;
+import static com.sonatype.nexus.api.common.NexusStringUtils.isNotBlank;
+import static hudson.Util.fixEmptyAndTrim;
+import static hudson.model.Result.FAILURE;
+import static hudson.util.FormValidation.error;
+import static hudson.util.FormValidation.ok;
+import static org.sonatype.nexus.ci.config.NxrmVersion.NEXUS_3;
+import static org.sonatype.nexus.ci.nxrm.Messages.CreateTag_DisplayName;
+import static org.sonatype.nexus.ci.nxrm.Messages.CreateTag_Error_TagAttributesJson;
+import static org.sonatype.nexus.ci.nxrm.Messages.CreateTag_Error_TagAttributesPath;
+import static org.sonatype.nexus.ci.nxrm.Messages.CreateTag_Validation_TagAttributesJson;
+import static org.sonatype.nexus.ci.nxrm.Messages.CreateTag_Validation_TagNameEmpty;
+import static org.sonatype.nexus.ci.util.RepositoryManagerClientUtil.nexus3Client;
+
+public class CreateTagStep
+    extends Builder
+    implements SimpleBuildStep
+{
+  static final Gson GSON = new Gson();
+
+  static final Type ATTRIBUTE_TYPE = new TypeToken<Map<String, Object>>() { }.getType();
+
+  private final String nexusInstanceId;
+
+  private final String tagName;
+
+  private String tagAttributesPath;
+
+  private String tagAttributesJson;
+
+  @DataBoundConstructor
+  public CreateTagStep(final String nexusInstanceId, final String tagName)
+  {
+    this.nexusInstanceId = checkArgument(nexusInstanceId, isNotBlank(nexusInstanceId), "Nexus Instance ID is required");
+    this.tagName = checkArgument(tagName, isNotBlank(tagName), "Tag Name is required");
```

Comment:
Trivial: Seems like both the messages could be added to the Messages.properties couldn't they?  I think both the move and delete will be able to use the same messages.